### DISCUSSION TITLE
Should vertico allow prompt selection when `vertico-preselect` is set to `first`?

### DISCUSSION THREAD

**Discussion by DevelopmentCool2449** [minad/vertico#488] [Q&A] [answered]
Hi, According to `vertico-preselect` doc:

> Configure if the prompt or first candidate is preselected. ...
> - first: Select the first candidate, allow prompt selection.

However when setting `vertico-preselect` to `'first` and using the arrow keys to up to prompt is not possible.

I found it's possible do it adding `(eq vertico-preselect 'first)` to `vertico--allow-prompt`
(https://github.com/minad/vertico/blob/main/vertico.el#L356)
```diff
@@ -355,6 +355,7 @@
       (vertico--hilit . ,(or hl #'identity))
       (vertico--allow-prompt . ,(and (not (eq vertico-preselect 'no-prompt))
                                      (or def-missing (eq vertico-preselect 'prompt)
+                                         (eq vertico-preselect 'first)
                                          (memq minibuffer--require-match
                                                '(nil confirm confirm-after-completion)))))
       (vertico--lock-candidate . ,lock)
```

That would make more easy for [submit a empty string](https://github.com/minad/vertico?tab=readme-ov-file#submitting-the-empty-string) for some commands instead using `vertico-exit-input`.

**Comment by minad** [comment]
`vertico-preselect=first` should allow prompt selection, but doesn't select the prompt by default. Note that prompt selection is only allowed for commands which support custom input (`REQUIRE-MATCH=nil`). See also the option `no-prompt` which completely disallows prompt selection.

**Reply by DevelopmentCool2449** [reply]
> `vertico-preselect=first` should allow prompt selection, but doesn't select the prompt by default. Note that prompt selection is only allowed for commands which support custom input (`REQUIRE-MATCH=nil`).

Then should be another option for `vertico-preselect`? for preselect first candidate and allow prompt selection even if `REQUIRE-MATCH=non-nil`, as a mix between `no-prompt` and `first`, the diff sent allow this for `first`.

**Reply by minad** [reply, marked as answer]
No, this would then give the impression that submitting arbitrary input is supported. We could however allow selecting the prompt only in the edge case where the prompt is empty. But then, I intentionally didn't implement this since I consider the concept of the "null completion" (submitting empty input) a design mistake. Many commands don't even support it - most famously `M-x`! Try `M-x M-RET`. This means to trigger the "mistake" you at least have to press `M-RET`.

**Reply by DevelopmentCool2449** [reply]
> No, this would then give the impression that submitting arbitrary input is supported. We could however allow selecting the prompt only in the edge case where the prompt is empty. But then, I intentionally didn't implement this since I consider the concept of the "null completion" (submitting empty input) a design mistake. Many commands don't even support it - most famously `M-x`! Try `M-x M-RET`. This means to trigger the "mistake" you at least have to press `M-RET`.

It makes sense, I was thinking if it's possible allow the null completion for some commands that allows it (even if `def` is not setted), icomplete-vertical does it for `M-x`, but i'm not sure if it's possible do it for Vertico.
Anyways thanks for the explanation.
